### PR TITLE
164489746 Users should be able to Login via Google, Facebook, Github or Twitter

### PR DESCRIPTION
**Pull request by frankhn**
#### What does this PR do?
  This PR allows a user to login with a social account.
#### Description of Task to be completed?

- [x] Creating a login component
- [x]   Creating social login buttons
- [x]   managing states with redux
- [x]   storing the user in local storage

#### How should this be manually tested?
  
- by logging in clicking social login buttons, 
- it should take you to a backend authentication
- then back to the frontend with the user token

#### Any background context you want to provide?
  
- we had to change the callback URL from the backend to match the frontend.

#### What are the relevant pivotal tracker stories?
  
- https://www.pivotaltracker.com/story/show/164489746

### CODE REVIEW

**Comment by istafrica-dennw** [comment]
@here 
@frankhn we shall only merge this PR if it is reviewed. tell you reviewers to review and make two mandatory.

**Review by mwibutsa — APPROVED** [review]
Great Job @frankhn, I know how you've been struggling to redirect the user back to the fronted but now You did it!

**Review by Inclet — CHANGES_REQUESTED** [review]
So far, Great Job! 
Here are a few things you need to consider:
- It would be better if you add a screenshot in your PR since we are also on front-end.
- Since I'm working on signup feature, we need also to have sync before merging this PR so we provide similar UI on the login form.

**Review by mystere10 — COMMENTED** [review]
Good frank, can I have a look at the screenshots? if the Heroku server is setup I would suggest you include the link in the PR description. thanks

File: src/components/SocialButtons.jsx
Comment (jnkindi) [inline src/components/SocialButtons.jsx:37]: Hello @frankhn , I would like to suggest that social login sections can be defined as standalone components and pass them here with their respective properties. for avoiding the similar blocks from ln 37 to ln 62. This will have a good impact on the code because the lines will be few.

**Review by jnkindi — CHANGES_REQUESTED** [review]

File: src/components/SocialButtons.jsx
Comment (frankhn) [reply src/components/SocialButtons.jsx:37]: Thanks for your feedback @jnkindi, basically, the social login is a component it's self and is called in Login component.

**Review by mwibutsa — CHANGES_REQUESTED** [review]
Implement those changes requested and re-raise your PR

**Pull request merged**
